util/execdetails/execdetails.go

```diff
@@ -512,37 +653,116 @@ func (e *RuntimeStatsWithConcurrencyInfo) SetConcurrencyInfo(infos ...*Concurren
 	}
 }
 
+// Clone implements the RuntimeStats interface.
+func (e *RuntimeStatsWithConcurrencyInfo) Clone() RuntimeStats {
+	newRs := &RuntimeStatsWithConcurrencyInfo{
+		concurrency: make([]*ConcurrencyInfo, 0, len(e.concurrency)),
+	}
+	newRs.concurrency = append(newRs.concurrency, e.concurrency...)
+	return newRs
+}
+
+// String implements the RuntimeStats interface.
 func (e *RuntimeStatsWithConcurrencyInfo) String() string {
 	var result string
-	if e.BasicRuntimeStats != nil {
-		result = fmt.Sprintf("time:%v, loops:%d", time.Duration(e.consume), e.loop)
-	}
 	if len(e.concurrency) > 0 {
-		for _, concurrency := range e.concurrency {
+		for i, concurrency := range e.concurrency {
+			if i > 0 {
+				result += ", "
+			}
 			if concurrency.concurrencyNum > 0 {
-				result += fmt.Sprintf(", %s:%d", concurrency.concurrencyName, concurrency.concurrencyNum)
+				result += fmt.Sprintf("%s:%d", concurrency.concurrencyName, concurrency.concurrencyNum)
 			} else {
-				result += fmt.Sprintf(", %s:OFF", concurrency.concurrencyName)
+				result += fmt.Sprintf("%s:OFF", concurrency.concurrencyName)
 			}
 		}
 	}
 	return result
 }
 
+// Merge implements the RuntimeStats interface.
+func (e *RuntimeStatsWithConcurrencyInfo) Merge(rs RuntimeStats) {
+	tmp, ok := rs.(*RuntimeStatsWithConcurrencyInfo)
+	if !ok {
+		return
+	}
+	e.concurrency = append(e.concurrency, tmp.concurrency...)
+}
+
 // RuntimeStatsWithCommit is the RuntimeStats with commit detail.
 type RuntimeStatsWithCommit struct {
-	RuntimeStats
 	Commit   *CommitDetails
 	LockKeys *LockKeysDetails
 }
 
+// Tp implements the RuntimeStats interface.
+func (e *RuntimeStatsWithCommit) Tp() int {
+	return TpRuntimeStatsWithCommit
+}
+
+// Merge implements the RuntimeStats interface.
+func (e *RuntimeStatsWithCommit) Merge(rs RuntimeStats) {
+	tmp, ok := rs.(*RuntimeStatsWithCommit)
+	if !ok {
+		return
+	}
+	if tmp.Commit != nil {
+		if e.Commit == nil {
+			e.Commit = &CommitDetails{}
+		}
+		e.Commit.Merge(tmp.Commit)
+	}
+
+	if tmp.LockKeys != nil {
+		if e.LockKeys == nil {
+			e.LockKeys = &LockKeysDetails{}
+		}
+		e.LockKeys.Merge(tmp.LockKeys)
+	}
+}
+
+// Clone implements the RuntimeStats interface.
+func (e *RuntimeStatsWithCommit) Clone() RuntimeStats {
+	newRs := RuntimeStatsWithCommit{}
+	if e.Commit != nil {
+		commit := &CommitDetails{
```

Comment:
Why not add a `Clone()` method for `CommitDetails` and `LockKeysDetails` for better maintainability? If we add a field to `CommitDetails` in the future, this unrelated function needs to be updated too.

Reply:
great, done.